vllm/model_executor/layers/quantization/torchao.py

```diff
@@ -0,0 +1,135 @@
+        return "torchao"
+
+    def get_supported_act_dtypes(self) -> List[torch.dtype]:
+        return [torch.float32, torch.float16, torch.bfloat16]
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        return 75
+
+    @staticmethod
+    def get_config_filenames() -> List[str]:
+        return ["config.json"]
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "TorchAOConfig":
+        """Create the quant config from an hf model config"""
+        from torchao.core.config import reconstruct_from_dict
+        hf_config = cls.get_from_keys_or(config, ["quant_type"], None)
+        runtime_quant = cls.get_from_keys_or(config, ["runtime_quant"], False)
+        assert hf_config is not None, "quant_type must be specified"
+        ao_config = reconstruct_from_dict(hf_config)
+        return cls(ao_config, runtime_quant)
+
+    def get_quant_method(self, layer: torch.nn.Module,
+                         prefix: str) -> Optional["TorchAOLinearMethod"]:
+        if isinstance(layer, LinearBase):
+            return TorchAOLinearMethod(self)
+        return None
+
+    def get_scaled_act_names(self) -> List[str]:
+        return []
+
+
+class TorchAOLinearMethod(LinearMethodBase):
+    """Linear method for torchao.
+
+    Args:
+        torchao_config: The torchao quantization config, a string
+        that encodes the type of quantization and all relevant arguments.
+    """
+
+    def __init__(self, quant_config: TorchAOConfig):
+        self.quant_config = quant_config
+
+    def create_weights(
+        self,
+        layer: torch.nn.Module,
+        input_size_per_partition: int,
+        output_partition_sizes: List[int],
+        input_size: int,
+        output_size: int,
+        params_dtype: torch.dtype,
+        **extra_weight_attrs,
+    ):
+        weight = Parameter(
+            torch.empty(
+                sum(output_partition_sizes),
+                input_size_per_partition,
+                dtype=params_dtype,
+            ),
+            requires_grad=False,
+        )
+        # we load quantized model in versions after 0.9.0 and do on the fly
+        # quantization in torchao versions before 0.9.0
+        if not self.quant_config.runtime_quant:
+            weight = torchao_quantize_param_data(
+                weight, self.quant_config.torchao_config)
+
+        set_weight_attrs(weight, {"input_dim": 1, "output_dim": 0})
+
+        layer.register_parameter("weight", weight)
+        set_weight_attrs(weight, extra_weight_attrs)
+
+    def process_weights_after_loading(self, layer: Module) -> None:
+        # We aren't loading a quantized weight, we quantize on the fly
+        if self.quant_config.runtime_quant:
+            torchao_config = self.quant_config.torchao_config
+            layer.weight = torchao_quantize_param_data(layer.weight,
+                                                       torchao_config)
+
+    # @torch.compile(mode="max-autotune-no-cudagraphs")
```

Comment:
Still need to figure out whats the best way to plum compile settings through

Reply:
Which `torch.compile` settings need to be passed through here?

Reply:
We have this helper which sets some global state: https://github.com/vllm-project/vllm/pull/14231/files#diff-797bf189c4ed30c72616d9da827c9c4534784128fe4d1a7e0ddd435d25009267R20

I want to ensure that this isn't being imported and ran by default 

Sidenote:
Is this pattern adding a torch.compile on the apply func correct?

Reply:
Most likely no. We have a model-wide torch compile wrapper so unless this is used outside the forward method or in models that don't support torch.compile, this is probably not right.

Could you elaborate on the use case here a bit more?

Reply:
Hey sorry I ended up adding specific tests for compile to ensure that it works accordingly and removed the intial line.

Most things appear to be working except that the inductor cache currently doens't have the correct AOTautograd calling wrapper to correctly unpack tensor subclasses. Spoke with Richard about this